src/js/pages/jobs/JobRunHistoryTable.js

```diff
@@ -233,24 +292,36 @@ class JobRunHistoryTable extends React.Component {
 
   render() {
     let {job} = this.props;
+    let {checkedItems} = this.state;
     let totalRunCount = job.getJobRuns().getItems().length;
+    let rightAlignLastNChildren = 0;
+    let hasCheckedTasks = Object.keys(checkedItems).length !== 0;
+    if (hasCheckedTasks) {
+      rightAlignLastNChildren = 1;
+    }
 
     return (
       <div>
-        <FilterHeadline
-          currentLength={totalRunCount}
-          inverseStyle={true}
-          name="Run"
-          onReset={function () {}}
-          totalLength={totalRunCount} />
-        <ExpandingTable
+        <FilterBar rightAlignLastNChildren={rightAlignLastNChildren}>
+          <FilterHeadline
+            currentLength={totalRunCount}
+            inverseStyle={true}
+            name="Run"
+            onReset={function () {}}
+            totalLength={totalRunCount} />
+          {this.getStopButton(hasCheckedTasks)}
+        </FilterBar>
+        <ExpandingWithCheckboxTable
+          allowMultipleSelect={false}
           className="job-run-history-table table table-hover inverse table-borderless-outer table-borderless-inner-columns flush-bottom"
           childRowClassName="job-run-history-table-child"
           columns={this.getColumns()}
```

Comment:
right here, you're passing columns into `ExpandingWithCheckboxTable` but then inside of that component, it has it's own `getColumns`. i think we can safely remove one right?